packages/graphback-cli/src/helpers/generate.ts

```diff
@@ -23,73 +25,68 @@ followed by ${chalk.cyan(`${cliName}db`)} to create database.
+    const customResolvers: string = join(process.cwd(), files.customResolvers, `custom`)
+    const generatedResolvers: string = join(process.cwd(), files.generatedResolvers, `generated`)
 
+    const backend: GraphQLBackendCreator = new GraphQLBackendCreator(schemaText, graphqlCRUD)
     const generated: IGraphQLBackend = await backend.createBackend(database)
-    
-    let generatedClient
-    let clientPath: string
-    if(client) {
-      generatedClient = await backend.createClient()
-      clientPath = `${process.cwd()}/client`
-      if(!existsSync(clientPath)) {
-        mkdirSync(clientPath)
-      }
-    }
 
-    writeFileSync(outputSchemaPath, generated.schema)
-
-    writeFileSync(`${outputResolverPath}/index.ts`, generated.resolvers.index)
-    
-    if(!existsSync(`${outputResolverPath}/custom`)) {
-      mkdirSync(`${outputResolverPath}/custom`)
-    }
-
-    if(!existsSync(`${outputResolverPath}/generated`)) {
-      mkdirSync(`${outputResolverPath}/generated`)
-    }
+    checkAndCreateFolders(pathForSchema, customResolvers, generatedResolvers);
 
     generated.resolvers.custom.forEach((output: OutputResolver) => {
-      if(!existsSync(`${outputResolverPath}/custom/${output.name}.ts`) || output.name === 'index') {
-        writeFileSync(`${outputResolverPath}/custom/${output.name}.ts`, output.output)
+      if (!existsSync(`${customResolvers}/${output.name}.ts`) || output.name === 'index') {
+        writeFileSync(`${customResolvers}/${output.name}.ts`, output.output)
       }
     })
 
-    const resolverFiles = readdirSync(`${outputResolverPath}/generated`)
-    resolverFiles.forEach((file: string) => unlinkSync(`${outputResolverPath}/generated/${file}`))
-
-    generated.resolvers.types.forEach((output: OutputResolver) => writeFileSync(`${outputResolverPath}/generated/${output.name}.ts`, output.output))
-    
-    if(client) {
+    writeFileSync(outputSchemaPath, generated.schema)
+    writeFileSync(`${generatedResolvers}/index.ts`, generated.resolvers.index)
+ 
+    generated.resolvers.types.forEach((output: OutputResolver) => writeFileSync(`${generatedResolvers}/${output.name}.ts`, output.output))
+
+    if (client) {
+      const generatedClient = await backend.createClient()
+      if (!existsSync(files.client)) {
+        mkdirSync(files.client, { recursive: true })
+      }
       Object.keys(generatedClient).forEach((folder: string) => {
-        const currentFolder = `${clientPath}/${folder}`
-        if(!existsSync(currentFolder)) {
+        const currentFolder = `${files.client}/${folder}`
+        if (!existsSync(currentFolder)) {
           mkdirSync(currentFolder)
         }
         generatedClient[folder].forEach((c: ClientImplementation) => writeFileSync(`${currentFolder}/${c.name}.ts`, c.implementation))
       })
     }
-
   } catch (err) {
     logError(err)
     process.exit(0)
   }
+
+  function checkAndCreateFolders(pathForSchema: string, customResolvers: string, generatedResolvers: string) {
```

Comment:
This could be made more generic:

```js
function checkAndCreateFolders(...paths: string[]) {
	for const path in paths {
       if (!existsSync(path)) {
          mkdirSync(path, { recursive: true });
       }
	}
}
```

Reply:
Good idea!

Reply:
Could throw an error mid loop if there were permissions problems

Reply:
So IMHO let's not change that but.. also try catch it. 
@darahayes is an walking node.js reference :)